src/jit/morph.cpp

```diff
@@ -12745,6 +12766,11 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                         oper = (oper == GT_LE) ? GT_EQ : GT_NE;
                         tree->SetOper(oper, GenTree::PRESERVE_VN);
                         tree->gtFlags &= ~GTF_UNSIGNED;
+
+                        if (op1->OperIs(GT_MOD))
```

Comment:
@EgorBo I was trying to review that PR and lost the track of the changes, the first part `Transform X MOD C == 0 to X UMOD C == 0` looks clear, but why were this label and goto added?

and it is really scary for me to have a back edge in a function like `fgMorphSmpOp` especially for 500 lines length.

Reply:
@sandreenko I wanted to handle `X % C != 0` but couldn't just visit `GT_NE` node - Roslyn emits `cgt.un` (GT_GT) and morph converts it back to GT_NE but doesn't visit GT_NE after. So I have three options to fix the != 0 case:
1) Handle unsigned GT_GT node instead of GT_NE
2) Convert GT_GT to GT_NE early in the importer 
3) goto GT_NE (current)

Reply:
Won't it be better to call `fgMorphSmpOp` recursively here?

Reply:
@sandreenko will try, but I think it there will be a few regressions - redundant long-to-int casts will appear.